src/main/java/com/teragrep/cfe_39/consumers/kafka/HDFSPrune.java

```diff
@@ -0,0 +1,139 @@
+            String hdfsuri = config.getHdfsuri(); // Get from config.
+
+            String path = config.getHdfsPath() + "/" + topicName;
+
+            // set kerberos host and realm
+            System.setProperty("java.security.krb5.realm", config.getKerberosRealm());
+            System.setProperty("java.security.krb5.kdc", config.getKerberosHost());
+
+            Configuration conf = new Configuration();
+
+            // enable kerberus
+            conf.set("hadoop.security.authentication", config.getHadoopAuthentication());
+            conf.set("hadoop.security.authorization", config.getHadoopAuthorization());
+
+            conf.set("fs.defaultFS", hdfsuri); // Set FileSystem URI
+            conf.set("fs.hdfs.impl", DistributedFileSystem.class.getName()); // Maven stuff?
+            conf.set("fs.file.impl", LocalFileSystem.class.getName()); // Maven stuff?
+
+            // hack for running locally with fake DNS records
+            // set this to true if overriding the host name in /etc/hosts
+            conf.set("dfs.client.use.datanode.hostname", config.getKerberosTestMode());
+
+            // server principal
+            // the kerberos principle that the namenode is using
+            conf.set("dfs.namenode.kerberos.principal.pattern", config.getKerberosPrincipal());
+
+            // set usergroup stuff
+            UserGroupInformation.setConfiguration(conf);
+            UserGroupInformation.loginUserFromKeytab(config.getKerberosKeytabUser(), config.getKerberosKeytabPath());
+
+            // filesystem for HDFS access is set here
+            fs = FileSystem.get(conf);
+
+            //==== Create folder if not exists
+            Path workingDir = fs.getWorkingDirectory();
+            newFolderPath = new Path(path);
+            if (!fs.exists(newFolderPath)) {
+                // Create new Directory
+                fs.mkdirs(newFolderPath);
+                LOGGER.info("Path " + path + " created.");
+            }
+        }
+        long pruneOffset = config.getPrune_offset();
+        cutoff_epoch = System.currentTimeMillis() - pruneOffset; // pruneOffset is parametrized in Config.java. Default value is 2 days in milliseconds.
+    }
+
+    public void prune() throws IOException {
+        // Fetch the filestatuses of HDFS files.
+        FileStatus[] fileStatuses = fs.listStatus(new Path(newFolderPath + "/"));
+        for (FileStatus a : fileStatuses) {
+            // If all the files have their modification timestamp altered to mirror the final record timestamp, it is possible to prune the database based on the timestamps of the fileStatuses object.
+            long convert = TimeUnit.MILLISECONDS.convert(a.getModificationTime(), TimeUnit.MICROSECONDS); // MICROSECONDS ARE NOT SUPPORTED, convert the microsecond epoch to milliseconds.
+            // Delete old files
+            if (convert < cutoff_epoch) {
+                boolean delete = fs.delete(a.getPath(), true);
```

Comment:
should there be a check in case the `newFolderPath` is empty and the path ends up being only "/" ?